### PR TITLE
Update Git.md

### PR DESCRIPTION
**Pull request by yin0409**
[+] add  whole git fetch information
[+] extra git fetch option but not all of the option.
[+] finishing one command with different method.
[+] each command have at one examples.
[+] each command contains full explanations.

### CODE REVIEW

**Comment by Maxlyu254** [comment]
You have some very detailed explanations of git fetch. However, you may also add some more descriptions to the first part of the page. In this way, it may allow readers to more easily understand how these commands work. I believe adding a little bit of explanation to git stash would help the readers a lot.

P.S. Look at line 45, there is a typo.

File: Topics/Development_Process/Git/Git.md [hunk 1]
```diff
@@ -33,4 +33,68 @@
 
         `git push origin master --force`
 
-        #Note: Use with precaution because we are using the force push. So try to limit the usage of this to directly push codes to the master branch. A great alternative would be using --force-with-lease because it does not overwrite any work on the master branch if more commits are added to the master branch. (In depth comparisons of force vs force-with-lease https://itnext.io/git-force-vs-force-with-lease-9d0e753e8c41)
\ No newline at end of file
+        #Note: Use with precaution because we are using the force push. So try to limit the usage of this to directly push codes to the master branch. A great alternative would be using --force-with-lease because it does not overwrite any work on the master branch if more commits are added to the master branch. (In depth comparisons of force vs force-with-lease https://itnext.io/git-force-vs-force-with-lease-9d0e753e8c41)
+
```
Comment (yi1z) [inline Topics/Development_Process/Git/Git.md:37]: Great explanations were added for the added git commands

File: Topics/Development_Process/Git/Git.md [hunk 2]
```diff
@@ -33,4 +33,68 @@
...
+   Compared with git pull and git Fetch:
+   same point is that available to accomplish the task.
+   different point is that git fetch is much safer than git pull.
+
+5.1: Wroks with remote branches
+
+To gain a deeper understanding of how git fetch operates, it's helpful to explore the way Git manages and archives commits. Within a Git repository, there is a hidden directory named `.git/objects`. This directory serves as the storage for all commits, encompassing both those from local and remote sources. To differentiate between commits from local and remote branches, Git utilizes branch references (refs). The refs associated with local branches can be found in the `.git/refs/heads/` directory. When you run the git branch command, it displays a list of these local branch refs.
+
+
```
Comment (yi1z) [inline Topics/Development_Process/Git/Git.md:49]: Examples are provided for how to use git commands to get the branches

File: Topics/Development_Process/Git/Git.md [hunk 3]
```diff
@@ -33,4 +33,68 @@
...
+   ```
+   /.git/refs/heads/
+   main
+   CSC207
+   CSC301
+   CSC209
+
+   ```
+
+
```
Comment (yi1z) [inline Topics/Development_Process/Git/Git.md:72]: This part explains the difference between local branches and remote branches, and I believe this is important for beginners to understand.

**Pull request merged**